src/GafferImage/OpenImageIOReader.cpp

```diff
@@ -0,0 +1,699 @@
+Gaffer::IntVectorDataPlug *OpenImageIOReader::availableFramesPlug()
+{
+	return getChild<IntVectorDataPlug>( g_firstPlugIndex + 3 );
+}
+
+const Gaffer::IntVectorDataPlug *OpenImageIOReader::availableFramesPlug() const
+{
+	return getChild<IntVectorDataPlug>( g_firstPlugIndex + 3 );
+}
+
+size_t OpenImageIOReader::supportedExtensions( std::vector<std::string> &extensions )
+{
+	std::string attr;
+	if( !getattribute( "extension_list", attr ) )
+	{
+		return extensions.size();
+	}
+
+	typedef boost::tokenizer<boost::char_separator<char> > Tokenizer;
+	Tokenizer formats( attr, boost::char_separator<char>( ";" ) );
+	for( Tokenizer::const_iterator fIt = formats.begin(), eFIt = formats.end(); fIt != eFIt; ++fIt )
+	{
+		size_t colonPos = fIt->find( ':' );
+		if( colonPos != string::npos )
+		{
+			std::string formatExtensions = fIt->substr( colonPos + 1 );
+			Tokenizer extTok( formatExtensions, boost::char_separator<char>( "," ) );
+			std::copy( extTok.begin(), extTok.end(), std::back_inserter( extensions ) );
+		}
+	}
+
+	return extensions.size();
+}
+
+void OpenImageIOReader::affects( const Gaffer::Plug *input, AffectedPlugsContainer &outputs ) const
+{
+	ImageNode::affects( input, outputs );
+
+	if( input == fileNamePlug() || input == refreshCountPlug() )
+	{
+		outputs.push_back( availableFramesPlug() );
+	}
+	
+	if( input == fileNamePlug() || input == refreshCountPlug() || input == missingFrameModePlug() )
+	{
+		for( ValuePlugIterator it( outPlug() ); it != it.end(); it++ )
+		{
+			outputs.push_back( it->get() );
+		}
+	}
+}
+
+void OpenImageIOReader::hash( const ValuePlug *output, const Context *context, IECore::MurmurHash &h ) const
+{
+	ImageNode::hash( output, context, h );
+
+	if( output == availableFramesPlug() )
+	{
+		fileNamePlug()->hash( h );
+		refreshCountPlug()->hash( h );
+	}
+}
+
+void OpenImageIOReader::compute( ValuePlug *output, const Context *context ) const
+{
+	if( output == availableFramesPlug() )
+	{
+		FileSequencePtr fileSequence = NULL;
+		IECore::ls( fileNamePlug()->getValue(), fileSequence, /* minSequenceSize */ 1 );
+
+		if( fileSequence )
+		{
+			IntVectorDataPtr resultData = new IntVectorData;
+			std::vector<FrameList::Frame> frames;
+			fileSequence->getFrameList()->asList( frames );
+			std::vector<int> &result = resultData->writable();
+			result.resize( frames.size() );
+			std::copy( frames.begin(), frames.end(), result.begin() );
+			static_cast<IntVectorDataPlug *>( output )->setValue( resultData );
+		}
+		else
+		{
+			static_cast<IntVectorDataPlug *>( output )->setToDefault();
+		}
+	}
+	else
+	{
+		ImageNode::compute( output, context );
+	}
+}
+
+void OpenImageIOReader::hashFileName( const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	// since fileName excludes frame substitutions
+	// but we internally vary the result output by
+	// frame, we need to explicitly hash the frame
+	// when the value contains FrameSubstitutions.
+	const std::string &fileName = fileNamePlug()->getValue();
```

Comment:
I'm surprised the compiler is allowing this, since `getValue()` returns a `string` rather than `string &`. I don't see anything to stop the string being destroyed before it gets hashed.